prbt_hardware_support/src/pilz_modbus_client.cpp

```diff
@@ -38,21 +38,22 @@ PilzModbusClient::PilzModbusClient(ros::NodeHandle& nh, const std::vector<unsign
 {
 }
 
-bool PilzModbusClient::init(const char* ip, unsigned int port, unsigned int retries, const ros::Duration& timeout)
+bool PilzModbusClient::init(const char* ip, unsigned int port, int retries, const ros::Duration& timeout)
 {
-  for (size_t retry_n = 0; retry_n < retries; ++retry_n)
+  size_t retry_n = 0;
+  while (ros::ok() && (retries == -1 || static_cast<int>(retry_n) < retries))
   {
+    ++retry_n;
+
     if (init(ip, port))
     {
       return true;
     }
 
-    ROS_ERROR_STREAM("Connection to " << ip << ":" << port << " failed. Try(" << retry_n + 1 << "/" << retries << ")");
+    ROS_WARN_STREAM_COND(retries == -1, "Connection to " << ip << ":" << port << " failed.");
```

Comment:
Add a hint here how to solve in the warning message (and check the output - I cannot remember this message during our tests at the real robot?):
https://github.com/PilzDE/pilz_robots/blob/a2ea32b49d1d431100604616fa8aa129916bc2a7/prbt_hardware_support/src/pilz_modbus_client_node.cpp#L127-L128

Reply:
Looks good, lets append it to the messages